### ISSUE TITLE
Invalid datetime string from MAPL_NetCDF.F90

### ISSUE THREAD

**Issue by metdyn** [GEOS-ESM/MAPL#2219]
Hi @darianboggs,

The NetCDF time units input from below generates an error in  MAPL_NetCDF.F90. 
```
 datetime_units = "seconds since 1970-01-01T00:00:00Z"
 call convert_NetCDF_DateTime_to_ESMF(int_time, datetime_units, interval, time0, time1=time1, tunit=tunit, _RC)

pe=00000 FAIL at line=00296    MAPL_NetCDF.F90                          <Invalid datetime string>
pe=00000 FAIL at line=00100    MAPL_NetCDF.F90                          <status=1>
```

I find no error with another format:
datetime_units = "seconds since 1970-01-01 00:00:00"

Please generalize the code. 
Thanks.

**Comment by darianboggs** [comment]
I'll take a look at this.

**Comment by metdyn** [comment]
An update from user request:
 - datetime_units = "seconds since 1993-1-1 00:00:00.0 0" ;
 - datetime_units = "seconds since 1970-01-01T00:00:00Z"

Thanks (@darianboggs).

**Comment by darianboggs** [comment]
> datetime_units = "seconds since 1993-1-1 00:00:00.0 0" ;
datetime_units = "seconds since 1970-01-01T00:00:00Z"

The second case is simple enough. An earlier revision of ISO 8601 allowed a space between the date and time portions of datetime strings. More recent revisions require a 'T' between the date and time portions. (This is for delimited strings. In undelimited strings, there is neither a space of 'T'.)

The first case is ill-formed. I'm surprised that NetCDF is using that.

I'll take a look at both.

**Comment by metdyn** [comment]
@darianboggs 
Besides the above two, I also encountered a string after Fortran-C interface calls  
datetime_units = "seconds since 1970-01-01T00:00:00Z#@"

Besides
datetime_units = "seconds since 1993-1-1 00:00:00.0 0" ;
datetime_units = "seconds since 1970-01-01T00:00:00Z"

I bypassed the issue with some small trick; hope you will have a systematic treatment!

**Comment by darianboggs** [comment]
Closed by #2170
